src/MultiDropdown.js

```diff
@@ -2,21 +2,84 @@ import React, { Component } from "react";
 import PropTypes from "prop-types";
 import * as countriesCities from "countries-cities";
 import "./MultiDropdown.css";
-import _ from "lodash";
-
-// HINT User CustomDropdown.js here
-
+import CustomDropdown from "./CustomDropdown";
 export default class MultiDropdown extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state ={
+        "selectedValue1":null,
+        "selectedValue2":null,
+      }
+  }
+  onChange(selectedValue,selectedElement) {
+    if(selectedValue){
+      if(selectedElement == "selectedValue1"){
+        this.onCountryChange(selectedValue.value);
+      }
+      else if(selectedElement == "selectedValue2"){
+        this.onCityChange(selectedValue.value);
+      }
+    }
+  }
+  clearValue (e) {
+		this.select.setInputValue('');
+	}
+  onCountryChange(selectedValue1){
+    this.setState({selectedValue1,selectedValue2:null});
+    this.props.onChange({selectedValue1,selectedValue2:null})
+  }
+  onCityChange(selectedValue2){
+    this.setState({selectedValue2});
+    this.props.onChange({selectedValue1:this.state.selectedValue1,selectedValue2})
+  }
+  getCities(){
+    if(this.state.selectedValue1){
+      let cityNames = countriesCities.getCities(this.state.selectedValue1).map(defaultValue=>{
+        return {
+          value:defaultValue,
+          label:defaultValue
+        }
+      });
+      return cityNames;
+    }
+    return [];
+  }
+  getCountries(){
+    let countryNames = countriesCities.getCountries().map(defaultValue=>{
+      return {
```

Comment:
This is an example of something that doesn't have to be in state. You're going to be doing this everytime there's a render - so this is an easy optimization.